contracts/protocols/mainnet/morpho/aave_v2/helpers.sol

```diff
@@ -0,0 +1,272 @@
+        uint256 avgBorrowRatePerYear; //in wad
+        uint256 p2pSupplyRate;
+        uint256 p2pBorrowRate;
+        uint256 poolSupplyRate;
+        uint256 poolBorrowRate;
+        uint256 totalP2PSupply;
+        uint256 totalPoolSupply;
+        uint256 totalP2PBorrows;
+        uint256 totalPoolBorrows;
+        uint256 p2pSupplyIndex;
+        uint256 p2pBorrowIndex;
+        uint256 poolSupplyIndex; //exchange rate of cTokens for compound
+        uint256 poolBorrowIndex;
+        uint256 updatedP2PSupplyIndex;
+        uint256 updatedP2PBorrowIndex;
+        uint256 updatedPoolSupplyIndex; //exchange rate of cTokens for compound
+        uint256 updatedPoolBorrowIndex;
+        uint256 lastUpdateTimestamp;
+        uint256 p2pSupplyDelta; //The total amount of underlying ERC20 tokens supplied through Morpho,
+        //stored as matched peer-to-peer but supplied on the underlying pool
+        uint256 p2pBorrowDelta; //The total amount of underlying ERC20 tokens borrow through Morpho,
+        //stored as matched peer-to-peer but borrowed from the underlying pool
+        uint256 reserveFactor;
+        uint256 p2pIndexCursor; //p2p rate position b/w supply and borrow rate, in bps,
+        // 0% = supply rate, 100% = borrow rate
+        AaveMarketDetail aaveData;
+        Flags flags;
+    }
+
+    struct Flags {
+        bool isCreated;
+        bool isPaused;
+        bool isPartiallyPaused;
+        bool isP2PDisabled;
+    }
+
+    struct UserMarketData {
+        MarketDetail marketData;
+        uint256 borrowRatePerYear;
+        uint256 supplyRatePerYear;
+        uint256 totalSupplies;
+        uint256 totalBorrows;
+        uint256 p2pBorrows;
+        uint256 p2pSupplies;
+        uint256 poolBorrows;
+        uint256 poolSupplies;
+        uint256 maxWithdrawable;
+        uint256 maxBorrowable;
+    }
+
+    struct UserData {
+        uint256 healthFactor; //calculated by updating interest accrue indices for all markets
+        uint256 collateralValue; //calculated by updating interest accrue indices for all markets
+        uint256 debtValue; //calculated by updating interest accrue indices for all markets
+        uint256 maxDebtValue; //calculated by updating interest accrue indices for all markets
+        bool isLiquidatable;
+        uint256 liquidationThreshold;
+        UserMarketData[] marketData;
+    }
+
+    IAaveLens internal aavelens = IAaveLens(0x507fA343d0A90786d86C7cd885f5C49263A91FF4);
+    IMorpho internal aaveMorpho = IMorpho(0x777777c9898D384F785Ee44Acfe945efDFf5f3E0);
+    IAave internal protocolData = IAave(getAaveProtocolDataProvider());
+    IAave internal incentiveData = IAave(getAaveIncentivesController());
+
+    function getLiquidatyData(
+        MarketDetail memory marketData_,
+        address poolTokenAddress_,
+        address asset
+    ) internal view returns (MarketDetail memory) {
+        (
+            ,
+            ,
+            ,
+            ,
+            ,
```

Comment:
@Richa-iitr What are these?

Reply:
Fetched all data in two methods due to stack too deep, the remaining details are fetched in the below method